gateway/src/apicast/policy/keycloak_role_check/keycloak_role_check.lua

```diff
@@ -0,0 +1,166 @@
+--- Keycloak Role Check Policy
+-- This policy verifies the realm roles and the client roles in the JWT.
+--
+--- The realm roles are specified when you want to add role check to the every client's resources or 3scale itself.
+-- https://www.keycloak.org/docs/4.0/server_admin/index.html#realm-roles
+--
+-- When you specify the realm roles in Keycloak, the JWT includes them as follows:
+--
+-- {
+--   "realm_access": {
+--     "roles": [
+--       "<realm_role_A>", "<realm_role_B>"
+--     ]
+--   }
+-- }
+--
+-- And you need to specify the "realm_roles" in this policy as follows:
+--
+-- "realm_roles": [
+--   { "name": "<realm_role_A>" }, { "name": "<realm_role_B>" }
+-- ]
+--
+--- The client roles are specified when you want to add role check to the particular client's resources.
+-- https://www.keycloak.org/docs/4.0/server_admin/index.html#client-roles
+--
+-- When you specify the client roles in Keycloak, the JWT includes them as follows:
+--
+-- {
+--   "resource_access": {
+--     "<client_A>": {
+--       "roles": [
+--         "<client_role_A>", "<client_role_B>"
+--       ]
+--     },
+--     "<client_B>": {
+--       "roles": [
+--         "<client_role_A>", "<client_role_B>"
+--       ]
+--     }
+--   }
+-- }
+--
+-- And you need to specify the "client_roles" in this policy as follows:
+--
+-- "client_roles": [
```

Comment:
Shouldn't this be called `resource_roles` ?

Reply:
I think `resource_roles` would be confusing since there is no resource role concept in RH-SSO. To add an item to the `resource_access` field of the access token, the administrator adds a *client role* to the *user roles*. 

I would stick to `client_roles`.

Reply:
I see. It is quite confusing for me because the property is called `resource_access` and it has no relation to `client_roles`. 

I know it is quite lengthy but `resource_access_client_roles` would work for me pretty well. But if I'm the only one then lets leave it :)